web/ui/react-app/src/api/Fetch.tsx

```diff
@@ -0,0 +1,75 @@
+import { Component } from 'react';
+import { fetchAPI } from './FetchAPI';
+
+export interface FetchState<T> {
+  data?: T;
+  error?: Error;
+}
+
+interface FetchProps {
+  url?: string;
+  urls?: string[];
+  options?: RequestInit;
+  children: (state: any) => any;
+}
+
+const urlsEqual = (urls?: string[], nextURLs?: string[]) => {
+  if (!urls || !nextURLs) {
+    return false;
+  }
+  return JSON.stringify(urls) === JSON.stringify(nextURLs);
+};
+
+export class Fetch extends Component<FetchProps, FetchState<any>> {
+  state = {
+    data: undefined,
+    error: undefined,
+  };
+
+  componentDidMount() {
+    const { url, urls, options } = this.props;
+    if (Boolean(url) && Boolean(urls)) {
```

Comment:
Stronger typing would make this check unnecessary

Reply:
and the ones below

Reply:
They have types ```url: string urls: string[]```, I'm just checking here whether user not passing both of them by mistake

Reply:
There should not be two ways to pass in urls. There should only be a `urls: string[]` field. Clients with one url can just pass it in as a single-element array.

Reply:
Yes, I already think about this and I came to a conclusion that is non-intuitive. Also, it brings more complexity in the code

Reply:
I disagree

Reply:
From user perspective or dev?

Reply:
Both, i think. As a dev, I have to maintain two options. As a user, I still have to read the code to understand what i'm supposed to pass in since there are two similar options.

Reply:
From a dev perspective, you should maintain  both options again since you don' t want to execute ```Promise.all``` logic for single URL. You have to check again the array whether it has a single element an if not - take ```Promise.all``` branch otherwise take the other direction...
From a user perspective I guess it would be easier, yes